flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1079,15 +1080,24 @@ private ApplicationReport startAppMaster(
 
         // setup security tokens
         if (UserGroupInformation.isSecurityEnabled()) {
-            // set HDFS delegation tokens when security is enabled
-            LOG.info("Adding delegation token to the AM container.");
-            List<Path> yarnAccessList =
-                    ConfigUtils.decodeListFromConfig(
-                            configuration, YarnConfigOptions.YARN_ACCESS, Path::new);
+            List<Path> yarnAccessList = new ArrayList<>();
```

Comment:
I would suggest to document that `yarn.security.kerberos.additionalFileSystems` should be unset  when obtaining delegation tokens are disabled.

Maybe we could also add a precondition check instead of process it silently.

Reply:
You mean that when `YARN_ACCESS` and `YARN_SECURITY_ENABLED` are existed together, Flink client should fast fail through throwing exception or exit Non-zero code.
Right? @wangyang0918

Reply:
If we disable obtaining the delegation tokens(HDFS/HBase), then `yarn.security.kerberos.additionalFileSystems` is expected to be unset. Right?
If this is the case, I think throwing exception here makes sense.

Reply:
Yes.
Agree with you, i will add precheck.

Reply:
Done. Please review it.